client/src/partials/journal/journal.js

```diff
@@ -29,26 +30,33 @@ JournalController.$inject = [
  *
  * @module bhima/controllers/JournalController
  */
-function JournalController(Transactions, Sorting, Grouping, Pagination, Filtering) {
+function JournalController(Transactions, Sorting, Grouping, Pagination, Filtering, ColumnConfig, AppCache, Modal) {
   var vm = this;
 
   // Journal utilites
-  var sorting, grouping, pagination, filtering;
+  var sorting, grouping, pagination, filtering, columnConfig;
+
+  //column list
+  var columns = null; 
+
+  //An appcache instance
+  var cache = AppCache('JournalGrid');
 
   // gridOptions is bound to the UI Grid and used to configure many of the
   // options, it is also used by the grid to expose the API
   vm.gridOptions = {};
 
   // Initialise each of the journal utilites, providing them access to the journal
   // configuration options
+  columnConfig = new ColumnConfig(vm.gridOptions); //This service should be responsible of initializing columns?
```

Comment:
I agree with this comment.

The `ColumnConfigService` shouldn't necessarily create all the columns, but definitely should be responsible for setting the `visibility` of each column based on the `cache`.  This would be similar to how the `FilteringService` and `GroupingService` work (see [here](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/js/services/journal/JournalGrouping.js#L120) for grouping's initialization, [here](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/js/services/journal/JournalSorting.js#L91) for sorting's init, and [here](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/js/services/journal/JournalFiltering.js#L26) for filtering).